Mathlib/Order/OrdContinuous.lean

```diff
@@ -0,0 +1,292 @@
+  fun s x h => by simpa only [image_image] using hg (hf h)
+#align left_ord_continuous.comp LeftOrdContinuous.comp
+
+-- PORTING NOTE: how to do this in non-tactic mode?
+protected theorem iterate {f : α → α} (hf : LeftOrdContinuous f) (n : ℕ) :
+    LeftOrdContinuous (f^[n]) :=
+by induction n with
+| zero => exact LeftOrdContinuous.id α
+| succ n ihn => exact ihn.comp hf
+
+#align left_ord_continuous.iterate LeftOrdContinuous.iterate
+
+end Preorder
+
+section SemilatticeSup
+
+variable [SemilatticeSup α] [SemilatticeSup β] {f : α → β}
+
+theorem map_sup (hf : LeftOrdContinuous f) (x y : α) : f (x ⊔ y) = f x ⊔ f y :=
+  (hf isLUB_pair).unique <| by simp only [image_pair, isLUB_pair]
+#align left_ord_continuous.map_sup LeftOrdContinuous.map_sup
+
+theorem le_iff (hf : LeftOrdContinuous f) (h : Injective f) {x y} : f x ≤ f y ↔ x ≤ y := by
+  simp only [← sup_eq_right, ← hf.map_sup, h.eq_iff]
+#align left_ord_continuous.le_iff LeftOrdContinuous.le_iff
+
+theorem lt_iff (hf : LeftOrdContinuous f) (h : Injective f) {x y} : f x < f y ↔ x < y := by
+  simp only [lt_iff_le_not_le, hf.le_iff h]
+#align left_ord_continuous.lt_iff LeftOrdContinuous.lt_iff
+
+variable (f)
+
+/-- Convert an injective left order continuous function to an order embedding. -/
+def toOrderEmbedding (hf : LeftOrdContinuous f) (h : Injective f) : α ↪o β :=
+  ⟨⟨f, h⟩, hf.le_iff h⟩
+#align left_ord_continuous.to_order_embedding LeftOrdContinuous.toOrderEmbedding
+
+variable {f}
+
+@[simp]
+theorem coe_to_order_embedding (hf : LeftOrdContinuous f) (h : Injective f) :
+    ⇑(hf.toOrderEmbedding f h) = f :=
+  rfl
+#align left_ord_continuous.coe_to_order_embedding LeftOrdContinuous.coe_to_order_embedding
+
+end SemilatticeSup
+
+section CompleteLattice
+
+variable [CompleteLattice α] [CompleteLattice β] {f : α → β}
+
+theorem map_Sup' (hf : LeftOrdContinuous f) (s : Set α) : f (supₛ s) = supₛ (f '' s) :=
+  (hf <| isLUB_supₛ s).supₛ_eq.symm
+#align left_ord_continuous.map_Sup' LeftOrdContinuous.map_Sup'
+
+theorem map_Sup (hf : LeftOrdContinuous f) (s : Set α) : f (supₛ s) = ⨆ x ∈ s, f x := by
+  rw [hf.map_Sup', supₛ_image]
+#align left_ord_continuous.map_Sup LeftOrdContinuous.map_Sup
+
+theorem map_supr (hf : LeftOrdContinuous f) (g : ι → α) : f (⨆ i, g i) = ⨆ i, f (g i) := by
+  simp only [supᵢ, hf.map_Sup', ← range_comp]
+  rfl
+#align left_ord_continuous.map_supr LeftOrdContinuous.map_supr
+
+end CompleteLattice
+
+section ConditionallyCompleteLattice
+
+variable [ConditionallyCompleteLattice α] [ConditionallyCompleteLattice β] [Nonempty ι] {f : α → β}
+
+theorem map_cSup (hf : LeftOrdContinuous f) {s : Set α} (sne : s.Nonempty) (sbdd : BddAbove s) :
+    f (supₛ s) = supₛ (f '' s) :=
+  ((hf <| is_lub_csupₛ sne sbdd).csupₛ_eq <| sne.image f).symm
+#align left_ord_continuous.map_cSup LeftOrdContinuous.map_cSup
+
+theorem map_csupr (hf : LeftOrdContinuous f) {g : ι → α} (hg : BddAbove (range g)) :
```

Comment:
```suggestion
theorem map_csupᵢ (hf : LeftOrdContinuous f) {g : ι → α} (hg : BddAbove (range g)) :
```
Similarly everywhere else. The `c` stays there, you can have a look [here](https://github.com/leanprover-community/mathlib4/blob/master/Mathlib/Order/ConditionallyCompleteLattice/Basic.lean) for various examples.